tests/qemu/test_qemu_backend.cpp

```diff
@@ -801,15 +802,24 @@ TEST_F(QemuBackend, createsQemuSnapshotsFromJsonFile)
     EXPECT_EQ(snapshot->get_parent(), parent);
 }
 
-TEST_F(QemuBackend, lists_no_networks)
+TEST_F(QemuBackend, networks_does_not_throw)
 {
     EXPECT_CALL(*mock_qemu_platform_factory, make_qemu_platform(_)).WillOnce([this](auto...) {
         return std::move(mock_qemu_platform);
     });
 
     mp::QemuVirtualMachineFactory backend{data_dir.path()};
 
-    EXPECT_THROW(backend.networks(), mp::NotImplementedOnThisBackendException);
+    auto [mock_platform, guard] = mpt::MockPlatform::inject();
+    EXPECT_CALL(*mock_platform, get_network_interfaces_info)
+        .WillOnce(Return(std::map<std::string, mp::NetworkInterfaceInfo>{
+            {"lxdbr0", {"lxdbr0", "bridge", "gobbledygook"}},
+            {"mpbr0", {"mpbr0", "bridge", "gobbledygook"}},
+            {"virbr0", {"virbr0", "bridge", "gobbledygook"}},
+            {"mpqemubr0", {"mpqemubr0", "bridge", "gobbledygook"}},
+            {"enxe4b97a832426", {"enxe4b97a832426", "ethernet", "gobbledygook"}}}));
+
+    EXPECT_NO_THROW(backend.networks());
```

Comment:
I do not think what you are trying to achieve is possible. 

The syntax `mock_qemu_platform->QemuPlatform::is_network_supported(arg)` is using the derived class (`MockQemuPlatform`) instance to call the base class (`QemuPlatform`) function implementation because of function hiding. This is useful when you want to call the base class function from a derived class instance. However, this only works if the base class has the function implementation and it is not the case here because `QemuPlatform::is_network_supported` is pure virtual. 

And there is no way it can somehow dispatch to `QemuPlatformDetail::is_network_supported`, it is a different derived class of `QemuPlatform` and `MockQemuPlatform` knows nothing about it, he only knows himself and the base class. 

Feel free to play with the [code snippet](https://godbolt.org/z/cd7qYnT14) to understand better.

Reply:
Yeah, I kinda thought as much. I wanted to do something like this in order to get some meaningful coverage on some of the new functions in the qemu platform.